projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_lineage_plugin.py

```diff
@@ -108,35 +109,46 @@ def _is_query_have_lineage(query_statement: str) -> bool:
         if query_statement is None:
             return False
 
-        statement_lines = query_statement.split(
-            os.linesep
-        )  # TODO if further optimization is needed, consider sqlparse
-        for line in statement_lines:
-            line = line.strip().lower()
-            if line.startswith("--") or (line.startswith("/*") and line.endswith("*/")):
-                continue  # Comments are omitted
-            if line.startswith("select 1 -- testing if connection is alive."):
-                return False  # managed_connection has a way to open and check connections with keep alive query
-            if line.startswith(
-                (
-                    "alter",
-                    "compute",
-                    "create",
-                    "describe",
-                    "drop",
-                    "explain",
-                    "grant",
-                    "invalidate",
-                    "refresh",
-                    "revoke",
-                    "set",
-                    "show",
-                    "truncate",
-                    "use",
-                )
-            ):
-                # these commands are not providing lineage data in the profile at the moment
-                return False
+        query_statement = query_statement.lower()
+        # managed_connection has a way to open and check connections with keep alive query
+        if "select 1 -- testing if connection is alive." in query_statement:
+            return False
+
+        query_statement = sqlparse.format(
```

Comment:
Why are you not using `sqlparse.parse` ?  

This will not work for CTEs : 
for example 
```
with x as (
   select * from y
)
select * from z
```
for example

Reply:
What will be the benefit of sqlparse.parse?
Mentioned example will work, already has this unit test